File: server/src/main/java/budgetpal/models/Account.java
package budgetpal.models;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@Table(name = "accounts")
public class Account {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    private String type;

    @Column(name = "account_number")
    private String accountNumber;
    
    private double balance;

    @Column(name = "is_active")
    private boolean isActive;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonBackReference(value = "accounts-users")
    private User user;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    public Account(String name, String type, String accountNumber, double balance, User user){
        this.name = name;
        this.type = type;
        this.accountNumber = accountNumber;
        this.balance = balance;
        this.isActive = true;
        this.user = user;
    }


}

File: mobile_client/lib/models/account.dart
class Account {
  final int id;
  final String name;
  final String type;
  final String accountNumber;
  final double balance;

  Account({
    required this.id,
    required this.name,
    required this.type,
    required this.accountNumber,
    required this.balance,
  });

  factory Account.fromJson(Map<String, dynamic> json) {
    return Account(
      id: json['id'],
      name: json['name'],
      type: json['type'],
      accountNumber: json['accountNumber'],
      balance: json['balance'],
    );
  }
}
